src/Zynga/Framework/IO/Web/V1/Manager.hh

```diff
@@ -50,4 +53,25 @@ class Manager {
       throw new UnexpectedHttpCodeException((string)$returnCode);
     }
   }
+  
+  public static function makeCurlRequest(
+    UrlBox $uploadUrl,
+    Map<string, mixed> $postParams): array<string, mixed> {
+      if(self::$useMockCurl == false) {
+        $session = curl_init();
+        curl_setopt( $session, CURLOPT_URL, $uploadUrl->get());
+        curl_setopt( $session, CURLOPT_POST, true);
+        curl_setopt( $session, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt( $session, CURLOPT_POSTFIELDS, http_build_query($postParams));
+        curl_setopt( $session, CURLOPT_TIMEOUT, 5);
+        $result = json_decode(curl_exec($session), true);
```

Comment:
Why not throw an exception here for failure? Hypothetically, we do not know if what the calling code should do if there is an  error for the curl call failing

Reply:
i am guessing you meant throw an exception when $result == null? Thats a valid use case and there is no reason to throw an exception in that case.

Reply:
To clarify, I meant that curl_exec will return false on failure. If it fails, we should let the calling code handle that case through an exception. Otherwise, we can do the nice thing and handle deserializing the return paramters for them.

http://php.net/manual/en/function.curl-exec.php